File: Programmers/노수일/Q42842.py
def solution(brown, yellow):
    division_numbers = []

    for i in range(1, int(yellow ** (1 / 2)) + 1):
        if yellow % i == 0:
            division_numbers.append((i, yellow // i))

    answer = []
    for division in division_numbers:
        square_width = (division[0] + 2) * (division[1] + 2)
        yellow_width = division[0] * division[1]

        if brown == square_width - yellow_width:
            answer.extend([division[1] + 2, division[0] + 2])
            break

    return answer

File: Programmers/유석환/Q42842.swift
import Foundation

func solution(_ brown:Int, _ yellow:Int) -> [Int] {
    let sum = brown + yellow
    var result = [Int]()
    
    for i in 3...sum {
        if sum % i == 0 {
            let j = sum / i
            if (i - 2) * (j - 2) == yellow {
                result = [i, j]
            }
        }
    }
    
    return result
}
